.eslintrc.json

```diff
@@ -40,28 +50,57 @@
       }
     ],
     "keyword-spacing": "error",
+    "linebreak-style": "error",
+    "max-statements-per-line": "error",
     "new-parens": "error",
     "no-array-constructor": "error",
+    "no-caller": "error",
     "no-console": "off",
+    "no-else-return": "error",
+    "no-eq-null": "error",
+    "no-eval": "error",
+    "no-extend-native": "error",
+    "no-extra-bind": "error",
+    "no-extra-label": "error",
     "no-extra-parens": "error",
     "no-floating-decimal": "error",
+    "no-implied-eval": "error",
+    "no-iterator": "error",
     "no-lone-blocks": "error",
     "no-lonely-if": "error",
     "no-multiple-empty-lines": "error",
     "no-multi-spaces": "error",
+    "no-multi-str": "error",
+    "no-native-reassign": "error",
+    "no-negated-condition": "error",
+    "no-new-wrappers": "error",
     "no-new-object": "error",
+    "no-octal-escape": "error",
     "no-path-concat": "error",
     "no-process-env": "error",
+    "no-proto": "error",
     "no-return-assign": "error",
+    "no-script-url": "error",
     "no-self-compare": "error",
     "no-sequences": "error",
+    "no-shadow-restricted-names": "error",
     "no-spaced-func": "error",
     "no-throw-literal": "error",
     "no-trailing-spaces": "error",
+    "no-undef-init": "error",
+    "no-unmodified-loop-condition": "error",
+    "no-unneeded-ternary": "error",
     "no-unused-expressions": "error",
+    "no-use-before-define": [
+      "error",
+      "nofunc"
```

Comment:
OK, just put it back then.

On Fri, Apr 8, 2016 at 7:28 PM, Alex Lam S.L. <notifications@github.com>
wrote:

> In .eslintrc.json
> <https://github.com/kangax/html-minifier/pull/614#discussion_r59051996>:
>
> >      "no-unused-expressions": "error",
> > +    "no-use-before-define": [
> > +      "error",
> > +      "nofunc"
>
> This is what happens without nofunc:
>
> src/htmlminifier.js
>    267:22  error  'minifyJS' was used before it was defined            no-use-before-define
>    285:14  error  'minifyURLs' was used before it was defined          no-use-before-define
>    298:14  error  'minifyCSS' was used before it was defined           no-use-before-define
>    302:12  error  'isMetaViewport' was used before it was defined      no-use-before-define
>    346:21  error  'minify' was used before it was defined              no-use-before-define
>    354:14  error  'minify' was used before it was defined              no-use-before-define
>    729:8   error  'minify' was used before it was defined              no-use-before-define
>   1157:13  error  'joinResultSegments' was used before it was defined  no-use-before-define
>
> src/htmlparser.js
>    90:30  error  'joinSingleAttrAssigns' was used before it was defined  no-use-before-define
>   168:42  error  'parseEndTag' was used before it was defined            no-use-before-define
>   174:29  error  'parseStartTag' was used before it was defined          no-use-before-define
>   177:11  error  'handleStartTag' was used before it was defined         no-use-before-define
>   194:26  error  'parseStartTag' was used before it was defined          no-use-before-define
>   232:7   error  'parseEndTag' was used before it was defined            no-use-before-define
>   242:5   error  'parseEndTag' was used before it was defined            no-use-before-define
>   271:7   error  'parseEndTag' was used before it was defined            no-use-before-define
>   276:9   error  'parseEndTag' was used before it was defined            no-use-before-define
>   281:7   error  'parseEndTag' was used before it was defined            no-use-before-define
>
> ? 18 problems (18 errors, 0 warnings)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kangax/html-minifier/pull/614/files/4a3bf237a2c9830457cbc6edad9db1a245eb0c90#r59051996>
>